src/zksync-specifics/examples/smart-account.md

```diff
@@ -0,0 +1,408 @@
+        bytes32,
+        bytes32,
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        bool success = _transaction.payToTheBootloader();
+        require(success, "Failed to pay the fee to the operator");
+    }
+
+    function prepareForPaymaster(
+        bytes32, // _txHash
+        bytes32, // _suggestedSignedHash
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        _transaction.processPaymasterInput();
+    }
+
+    fallback() external {
+        // fallback of default account shouldn't be called by bootloader under no circumstances
+        assert(msg.sender != BOOTLOADER_FORMAL_ADDRESS);
+
+        // If the contract is called directly, behave like an EOA
+    }
+
+    receive() external payable {
+        // If the contract is called directly, behave like an EOA.
+        // Note, that is okay if the bootloader sends funds with no calldata as it may be used for refunds/operator payments
+    }
+}
+```
+
+### `DeployMultisig`
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "forge-std/Script.sol";
+import "@era-contracts/libraries/SystemContractsCaller.sol";
+import {Create2Factory} from "@era-contracts/Create2Factory.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../src/AAFactory.sol";
+import "../src/TwoUserMultisig.sol";
+
+contract DeployMultisig is Script {
+    function run() external {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+
+        // Owners for the multisig account
+        // Can be random
+        address owner1 = vm.envAddress("OWNER_1");
+        address owner2 = vm.envAddress("OWNER_2");
+
+        // Read artifact file and get the bytecode hash
+        string memory artifact = vm.readFile(
+            "zkout/TwoUserMultisig.sol/TwoUserMultisig.json"
+        );
+        bytes32 multisigBytecodeHash = vm.parseJsonBytes32(artifact, ".hash");
+        console.log("Bytecode hash: ");
+        console.logBytes32(multisigBytecodeHash);
+        bytes32 salt = "1234";
+
+        vm.startBroadcast(deployerPrivateKey);
+        AAFactory factory = new AAFactory(multisigBytecodeHash);
+        console.log("Factory deployed at: ", address(factory));
+
+        // Mark the bytecode as a factory dependency
+        vmExt.zkUseFactoryDep("TwoUserMultisig");
+
+        factory.deployAccount(salt, owner1, owner2);
+        string memory factoryArtifact = vm.readFile(
+            "zkout/AAFactory.sol/AAFactory.json"
+        );
+        bytes32 factoryBytecodeHash = vm.parseJsonBytes32(
+            factoryArtifact,
+            ".hash"
+        );
+        Create2Factory create2Factory = new Create2Factory();
+        address multisigAddress = create2Factory.create2(
+            salt,
+            factoryBytecodeHash,
+            abi.encode(owner1, owner2)
+        );
+        console.log("Multisig deployed at: ", multisigAddress);
```

Comment:
I don't understand this section - we just deployed `AAFactory` and even used it, but now we do the same? But with a different constructor even?
Above we have `bytecodeHash` and here it's `owner1,owner2` encoded...
Also technically speaking we can already make use of the `Create2Factory` implicitly with the CREATE2 syntax ;)